sshconnect.c

```diff
@@ -469,19 +470,20 @@ ssh_connect_direct(struct ssh *ssh, const char *host, struct addrinfo *aitop,
 		 */
 		for (ai = aitop; ai; ai = ai->ai_next) {
 			if (ai->ai_family != AF_INET &&
-			    ai->ai_family != AF_INET6) {
+			    ai->ai_family != AF_INET6 &&
+			    ai->ai_family != AF_UNIX) {
 				errno = EAFNOSUPPORT;
 				continue;
 			}
-			if (getnameinfo(ai->ai_addr, ai->ai_addrlen,
+			if (ai->ai_family != AF_UNIX && getnameinfo(ai->ai_addr, ai->ai_addrlen,
 			    ntop, sizeof(ntop), strport, sizeof(strport),
```

Comment:
getnameinfo supports AF_UNIX on glibc. It returns localhost in ntop and the path in strport.

Reply:
Perhaps the syntax to connect to a unix socket should be "ssh localhost -p /path/to/socket"? It makes some sense that the path goes into the "port" part instead of the host.

Reply:
> getnameinfo supports AF_UNIX on glibc. 

Hm, maybe not on MacOS, and that's why I changed it. I'll check that later.

> Perhaps the syntax to connect to a unix socket should be "ssh localhost -p /path/to/socket"? It makes some sense that the path goes into the "port" part instead of the host part.

I think it would be confusing. It's not used that way for the `-L` and `-R` options. And I am not sure how that would work for the `HostAlias` option.

Reply:
Yes, you are probably right. We should never talk about host/port when it comes to AF_UNIX. So that's why relying on getnameinfo is bad even if it happen to support AF_UNIX. Resolving this discussion thread.

Reply:
I made small change so it prints correctly prints out "Connecting to..." when debugging.